### DISCUSSION TITLE
Purpur API Addition - Block Break Speed

### DISCUSSION THREAD

**Discussion by kohlerpop1** [PurpurMC/Purpur#1302] [__Dismissed__] [answered]
Preface
---
I originally requested this as a feature in Paper, but they kind of blew me off/told me no. However, I think this would be a good addition to a more feature fun related Purpur so I decided to open a request here instead.

[Here is a link to the original issue if you wish to view it!](https://github.com/PaperMC/Paper/issues/8192)

Altered Post Content:
---
### Is your feature request related to a problem?
I would like a small addition to the purpur API to be able to edit the breaking speed of a specific block/material in the server. I am using blocks which are retextured from mushroom/noteblock blocks but I am unable to change the breaking speed of the mushrooms/noteblocks to match the material that is being broken.

### Describe the solution you'd like.
I know that it is possible to do this with NMS so I was wondering if it would be possible to add it to the API to be able to access/change the values. I have looked into this and have found a few ways of doing it. I am currently using NMS but would like to see an addition to the Purpur API if possible.
Here are a few references and code examples.
[Example 1](https://www.spigotmc.org/threads/change-break-speed.148002/)
[Example 2](https://bukkit.org/threads/change-block-break-time.299960/)
[Example 3](https://www.spigotmc.org/threads/1-8-1-13-custom-block-breaking-change-block-hardness.362586/)

### Describe alternatives you've considered.
I have tried all of these and it currently is working it is just a pain for it to be using NMS and I believe would be a nice addition to the API.
I have also tried using spell effects to slow down there breaking speed but that just throws off interactions and gameplay.

**Comment by RhythmicSys** [comment, marked as answer]
In my experience in attempting to mess with altering break speed, it causes very strange desync issues and causing unexpected results like the player being unable to hear blocks break, and blocks only breaking at the expected speed if they were under a certain hardness, otherwise the changed blockbreak speed didn't change at all. I'm really not sure this is a plausible addition without it just causing major issues.

**Reply by kohlerpop1** [reply]
Yeah I am aware it can cause desync issues with both the client and server, however, it is still better than having nothing at all. I propose it be added and remain experimental as people already do it and cause desync issues using NMS and more!

**Reply by 1ByteBit** [reply]
Just because people do it knowing the risks doesn't mean purpur should allow more people to do it and take those risks @kohlerpop1

**Reply by kohlerpop1** [reply]
Valid point. However, I currently am forced to use and need it therefore I believe it would be very nice to have and use! @1ByteBit
